### PR TITLE
Seek 1.13 linked seek custom metadata

### PR DESCRIPTION
**Pull request by whomingbird**

### CODE REVIEW

File: app/models/custom_metadata.rb [hunk 1]
```diff
@@ -3,11 +3,33 @@ class CustomMetadata < ApplicationRecord
...
+  def update_linked_custom_metadata_id
+    linked_custom_metadatas.each do |cm|
+      attr_name = cm.custom_metadata_attribute.title
+      data.mass_assign(data.to_hash.update({attr_name => cm.id}), pre_process: false)
+      update_column(:json_metadata, data.to_json)
+    end
+  end
+
+  def has_linked_custom_metadatas?
+    !linked_custom_metadatas.blank?
```
Comment (stuzart) [inline app/models/custom_metadata.rb:29]: you can just use `any?`

File: app/helpers/samples_helper.rb [hunk 2]
```diff
@@ -109,6 +145,24 @@ def seek_cv_attribute_display(value, attribute)
     content
   end
 
+  def linked_custom_metadata_attribute_display(value)
+    html = ''
+     data = CustomMetadata.find(value.id).data
+    html += '<ul>'
+       CustomMetadata.find(value.id).custom_metadata_attributes.each do |attr|
```
Comment (stuzart) [inline app/helpers/samples_helper.rb:152]: double calls to load `CustomMetadata.find(value.id)` - can just do once and store in a variable

File: lib/seek/json_metadata/serialization.rb [hunk 3]
```diff
@@ -18,9 +18,13 @@ def data
       end
 
       def get_attribute_value(attr)
-        attr = attr.accessor_name if attr.is_a?(attribute_class)
-
-        data[attr.to_s]
+        if attr.try(:sample_attribute_type) && attr.sample_attribute_type.base_type == Seek::Samples::BaseType::LINKED_CUSTOM_METADATA
```
Comment (stuzart) [inline lib/seek/json_metadata/serialization.rb:21]: you can use `linked_custom_metadata?` here, e.g `attr.try(:sample_attribute_type).try(:linked_custom_metadata?)`

Comment (stuzart) [reply lib/seek/json_metadata/serialization.rb:21]: also, what if `get_attribute_value `is called for linked_custom_metadata but `attr `is a string ?

File: lib/seek/samples/attribute_type_handlers/linked_custom_metadata_attribute_type_handler.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    module AttributeTypeHandlers
+      class LinkedCustomMetadataAttributeTypeHandler < BaseAttributeHandler
+
+        def test_value(array_value)
+
+        end
+
+        def convert(value)
+
+        end
```
Comment (stuzart) [inline lib/seek/samples/attribute_type_handlers/linked_custom_metadata_attribute_type_handler.rb:12]: this looks wrong, they override the parent class methods, but will just return nil

**Review by stuzart — CHANGES_REQUESTED** [review]

File: lib/seek/json_metadata/serialization.rb (hunk 3 above)
Comment (whomingbird) [reply lib/seek/json_metadata/serialization.rb:21]: > also, what if get_attribute_value is called for linked_custom_metadata but attr is a string ?


when `get_attribute_value` is called by `show` action, the value output is
```
attr=>#<CustomMetadataAttribute id: 286, custom_metadata_type_id: 16, sample_attribute_type_id: 26, required: false, pos: nil, title: "a_3_3_3", sample_controlled_vocab_id: nil, description: nil, label: nil, linked_custom_metadata_type_id: 15>
attr.linked_custom_metadata?=>true
value=>#<CustomMetadata id: 14, json_metadata: "{\"a_3_3_3_1\":\"3331\",\"a_3_3_3_2\":\"3332\",\"a_3_3_3_3\"...", item_type: nil, item_id: nil, custom_metadata_type_id: 15, custom_metadata_attribute_id: 286>
```

when `get_attribute_value` is called by `edit` action, the value output is
```
attr=>"a_3_3_3"
self=>#<CustomMetadata id: 13, json_metadata: "{\"a_3_3_1\":\"331\",\"a_3_3_2\":\"332\",\"a_3_3_3\":14}", item_type: nil, item_id: nil, custom_metadata_type_id: 16, custom_metadata_attribute_id: 296>
value=>14
```

**Review by stuzart — APPROVED** [review]

**Comment by stuzart** [comment]
merged, but not automatically detected as conflicts were detected

**Pull request closed without merging**